peers/open_channels.js

```diff
@@ -237,56 +304,85 @@ module.exports = (args, cbk) => {
-
-          if (!node.sockets.length) {
-            return cbk([503, 'NoAddressFoundToConnectToNode', {node}]);
-          }
-
-          args.logger.info({
-            connecting_to: {alias: node.alias, public_key: node.public_key},
-          });
+        return cbk();
+      }],
 
-          return asyncRetry({times}, cbk => {
-            return asyncDetectSeries(node.sockets, ({socket}, cbk) => {
-              return addPeer({socket, lnd: args.lnd, public_key: key}, err => {
-                return cbk(null, !err);
-              });
-            },
-            (err, res) => {
-              if (!!err) {
-                return cbk(err);
+      // Get peers, check if already connected and connect if required
+      connect: ['getNodes', ({getNodes}, cbk) => {
+        return asyncEach((getNodes), (node, cbk) => {
+          return getPeers({lnd: node.opening_node_lnd}, (err, res) => {
+            if (!!err) {
+              return cbk(err);
+            }
+              // Exit early when the peer is already connected
+              if (res.peers.map(n => n.public_key).includes(node.public_key)) {
+                return cbk();
+              }
+    
+              if (!node.sockets.length) {
+                return cbk([503, 'NoAddressFoundToConnectToNode', {node}]);
               }
 
-              if (!res) {
-                return cbk([503, 'FailedToConnectToPeer', ({peer: key})]);
+            if (!!args.opening_nodes && !!args.opening_nodes.length) {
+                  args.logger.info({
+                    connecting_to: {alias: node.alias, public_key: node.public_key, from: node.opening_node_name},
+                  });
+              } else {
+                  args.logger.info({
+                  connecting_to: {alias: node.alias, public_key: node.public_key},
+                });
               }
 
-              return setTimeout(() => cbk(null, true), peerAddedDelayMs);
-            });
-          },
-          cbk);
+              return asyncRetry({times}, cbk => {
+                return asyncDetectSeries(node.sockets, ({socket}, cbk) => {
+                  return addPeer({socket, lnd: node.opening_node_lnd, public_key: node.public_key}, err => {
+                    return cbk(null, !err);
+                  });
+                },
+                (err, res) => {
+                  if (!!err) {
+                    return cbk(err);
+                  }
+    
+                  if (!res) {
+                    return cbk([503, 'FailedToConnectToPeer', ({peer: node.public_key})]);
+                  }
+    
+                  return setTimeout(() => cbk(null, true), peerAddedDelayMs);
+                });
+              },
+              cbk);
+          });
         },
         cbk);
       }],
 
+      // Get the wallet version and check if it is compatible
+      getWalletVersion: ['validate', ({}, cbk) => {
+        return getWalletVersion({lnd: args.lnd}, (err, res) => {
```

Comment:
This call could be taken out now completely because all supported versions should be compatible